### ISSUE TITLE
Precision for multilabel

### ISSUE THREAD

**Issue by josianerodrigues** [flyingpot/pytorch_deephash#8]
Hi @flyingpot,
I'm sorry to bother you again. I modified the model to train on a multilabel base, but I'm having difficulty calculating the map. Your function that calculates map is this here:

```
for i in range(query_times):
        print('Query ', i+1)
        query_label = tst_label[i]
        query_binary = tst_binary[i,:]
        query_result = np.count_nonzero(query_binary != trn_binary, axis=1)    #don't need to divide binary length
        sort_indices = np.argsort(query_result)
        buffer_yes= np.equal(query_label, trn_label[sort_indices]).astype(int)
        P = np.cumsum(buffer_yes) / Ns
        AP[i] = np.sum(P * buffer_yes) /sum(buffer_yes)
    map = np.mean(AP)
    print(map)
```

How would that line of code look `buffer_yes= np.equal(query_label, trn_label[sort_indices]).astype(int)` to return relevant if it has at least one correct label? The image label consists of an array of 0s and 1s of size 21 where 1 corresponds to that label exists for that image. Thank you so much.

**Comment by josianerodrigues** [comment]
I already got it here, thank you.

**Comment by josianerodrigues** [comment]
Hi @flyingpot,
I have a question regarding the network structure to train a multilabel dataset and maybe you know how to respond.
For a multiclass classification the end of the network is thus, according to its implementation.

```
        self.Linear1 = nn.Linear(4096, self.bits)
        self.sigmoid = nn.Sigmoid()
        self.Linear2 = nn.Linear(self.bits, 10)

```
I tried to train the network a multilabel classification with this same setting just by changing the number of classes and the loss function for MultiLabelSoftMarginLoss, but this did not work. Do I need to change anything else?

**Comment by josianerodrigues** [comment]
I also tested the loss function BCEWithLogitsLoss, but it did not work too.
